include/alpaka/dev/DevUniformCudaHipRt.hpp

```diff
@@ -43,7 +45,20 @@ namespace alpaka
     {
         template<typename TApi, bool TBlocking>
         class QueueUniformCudaHipRt;
-    }
+
+        inline auto getThreadPool() -> core::ThreadPool&
+        {
+#    if BOOST_COMP_CLANG
+#        pragma clang diagnostic push
+#        pragma clang diagnostic ignored "-Wexit-time-destructors"
+#    endif
+            static core::ThreadPool pool;
```

Comment:
I am confused, I thought we discussed that the thread pool will be part of the device and not unbound to any object.
Take care in cases where a device is destroyed you must check that the thread pool has no outstanding tasks else it could be you executing on invalid devices or memory.

IMO the pool should be a member of `DevUniformCudaHipRt`

Reply:
It keeps alpaka global state free and avoids issue if a device got [reset](https://github.com/alpaka-group/alpaka/blob/0819e45098d38dc3c62a3fcee1e513e0492658d9/include/alpaka/dev/DevUniformCudaHipRt.hpp#L161) by the user.

During think about the issue, I checked the code and realized that alpaka is not currently freeing the device.